core/src/main/java/org/apache/druid/data/input/SegmentsSplitHintSpec.java

```diff
@@ -56,6 +57,12 @@ public long getMaxInputSegmentBytesPerTask()
     return maxInputSegmentBytesPerTask;
   }
 
+  @Override
+  public <T> Iterator<List<T>> split(Iterator<T> inputIterator, Function<T, InputFileAttribute> inputAttributeExtractor)
```

Comment:
Seems like this method really doesn't belong here if not all subclasses or implementation need it? Or should this class be abstract instead?

Reply:
Added comment about it.